views/default/navigation/pagination.php

```diff
@@ -37,103 +37,118 @@
 	$base_url = current_page_url();
 }
 
-$num_pages = elgg_extract('num_pages', $vars, 10);
-$delta = ceil($num_pages / 2);
-
 if ($count <= $limit && $offset == 0) {
 	// no need for pagination
 	return true;
 }
 
-$total_pages = ceil($count / $limit);
-$current_page = ceil($offset / $limit) + 1;
-
-$pages = new stdClass();
-$pages->prev = array(
-	'text' => elgg_echo('previous'),
-	'href' => '',
-	'is_trusted' => true,
-);
-$pages->next = array(
-	'text' => elgg_echo('next'),
-	'href' => '',
-	'is_trusted' => true,
-);
-$pages->items = array();
-
-// Add pages before the current page
-if ($current_page > 1) {
-	$prev_offset = $offset - $limit;
-	if ($prev_offset < 1) {
-		// don't include offset=0
-		$prev_offset = null;
-	}
+$total_pages = (int) ceil($count / $limit);
+$current_page = (int) ceil($offset / $limit) + 1;
 
-	$pages->prev['href'] = elgg_http_add_url_query_elements($base_url, array($offset_key => $prev_offset));
+$pages = array();
 
-	$first_page = $current_page - $delta;
-	if ($first_page < 1) {
-		$first_page = 1;
-	}
+// determine starting page
+$start_page = max(min([sanitize_int($current_page - 2, true), sanitize_int($total_pages - 4, true)]), 1);
```

Comment:
Ugh. We need Math::abs or something. Lol what a secretive behavior based on
a boolean...

On Fri, Mar 13, 2015, 3:39 AM Jeroen Dalsem <notifications@github.com>
wrote:

> In views/default/navigation/pagination.php
> <https://github.com/Elgg/Elgg/pull/8051#discussion_r26376513>:
>
> >
> > -	$first_page = $current_page - $delta;
> > -	if ($first_page < 1) {
> > -		$first_page = 1;
> > -	}
> > +// determine starting page
> > +$start_page = max(min([sanitize_int($current_page - 2, true), sanitize_int($total_pages - 4, true)]), 1);
>
> it makes positive ints (with the true flag)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/8051/files#r26376513>.
>